arangod/Aql/ClusterNodes.cpp

```diff
@@ -589,21 +589,35 @@ struct ParallelizableFinder final : public WalkerWorker<ExecutionNode> {
   }
 
   bool before(ExecutionNode* node) override final {
-    if (node->getType() == ExecutionNode::SCATTER || node->getType() == ExecutionNode::GATHER ||
-        node->getType() == ExecutionNode::DISTRIBUTE ||
-        node->getType() == ExecutionNode::TRAVERSAL ||
-        node->getType() == ExecutionNode::SHORTEST_PATH ||
-        node->getType() == ExecutionNode::K_SHORTEST_PATHS) {
+    LOG_DEVEL << "ParallelizableFinder::before " << node->id();
+    auto nodeType = node->getType();
+
+    if (nodeType == ExecutionNode::SCATTER || nodeType == ExecutionNode::GATHER ||
+        nodeType == ExecutionNode::DISTRIBUTE) {
+      LOG_DEVEL << "not parallelizable (1)";
       _isParallelizable = false;
       return true;  // true to abort the whole walking process
     }
+
+    if (nodeType == ExecutionNode::TRAVERSAL || nodeType == ExecutionNode::SHORTEST_PATH ||
+        nodeType == ExecutionNode::K_SHORTEST_PATHS) {
+      LOG_DEVEL << "traversal node";
+      GraphNode* gn = ExecutionNode::castTo<GraphNode*>(node);
+      if (!gn->isUsedAsSatellite()) {
```

Comment:
As discussed, it suffices to test whether the `GraphNode` is also a `SatelliteGraphNode`. e.g
```suggestion
      auto* gn = ExecutionNode::castTo<GraphNode*>(node);
      if (!gn->isSatelliteNode()) {
```